src/common/regions/utils.ts

```diff
@@ -0,0 +1,33 @@
+import { concat, partition } from 'lodash';
+import regions from './region_db';
+import { getStateFips } from './regions_data';
+import { County, State, Region, MetroArea } from './types';
+
+export function belongsToState(county: County, stateFips: string | null) {
+  return county.state.fipsCode === stateFips;
+}
+
+export function getAutocompleteLocations(region?: Region) {
+  const { states, metroAreas, counties } = regions;
+
+  // Homepage
+  if (!region) {
+    return concat<Region>(states, metroAreas, counties);
+  }
+
+  // Location pages
+  if (region instanceof MetroArea) {
+    const [countiesInMetro, otherCounties] = partition(counties, county =>
+      region.counties.includes(county),
+    );
+    return concat<Region>(countiesInMetro, states, otherCounties, metroAreas);
+  } else if (region instanceof State || region instanceof County) {
+    const stateFips = getStateFips(region);
+    const [stateCounties, otherCounties] = partition(counties, county =>
+      belongsToState(county, stateFips),
+    );
+    return concat<Region>(stateCounties, states, metroAreas, otherCounties);
```

Comment:
Happy to fix this up in a separate PR, but I would love to partition the metros to pull out metros that have counties in the state